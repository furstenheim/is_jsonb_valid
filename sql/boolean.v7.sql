-- boolean schema 'true'
-- number is valid
SELECT is_jsonb_valid_draft_v7('true', '1');
-- string is valid
SELECT is_jsonb_valid_draft_v7('true', '"foo"');
-- boolean true is valid
SELECT is_jsonb_valid_draft_v7('true', 'true');
-- boolean false is valid
SELECT is_jsonb_valid_draft_v7('true', 'false');
-- null is valid
SELECT is_jsonb_valid_draft_v7('true', 'null');
-- object is valid
SELECT is_jsonb_valid_draft_v7('true', '{"foo":"bar"}');
-- empty object is valid
SELECT is_jsonb_valid_draft_v7('true', '{}');
-- array is valid
SELECT is_jsonb_valid_draft_v7('true', '["foo"]');
-- empty array is valid
SELECT is_jsonb_valid_draft_v7('true', '[]');
-- boolean schema 'false'
-- number is invalid
SELECT is_jsonb_valid_draft_v7('false', '1');
-- string is invalid
SELECT is_jsonb_valid_draft_v7('false', '"foo"');
-- boolean true is invalid
SELECT is_jsonb_valid_draft_v7('false', 'true');
-- boolean false is invalid
SELECT is_jsonb_valid_draft_v7('false', 'false');
-- null is invalid
SELECT is_jsonb_valid_draft_v7('false', 'null');
-- object is invalid
SELECT is_jsonb_valid_draft_v7('false', '{"foo":"bar"}');
-- empty object is invalid
SELECT is_jsonb_valid_draft_v7('false', '{}');
-- array is invalid
SELECT is_jsonb_valid_draft_v7('false', '["foo"]');
-- empty array is invalid
SELECT is_jsonb_valid_draft_v7('false', '[]');