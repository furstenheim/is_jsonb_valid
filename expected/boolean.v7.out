-- boolean schema 'true'
-- number is valid
SELECT is_jsonb_valid_draft_v7('true', '1');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- string is valid
SELECT is_jsonb_valid_draft_v7('true', '"foo"');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- boolean true is valid
SELECT is_jsonb_valid_draft_v7('true', 'true');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- boolean false is valid
SELECT is_jsonb_valid_draft_v7('true', 'false');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- null is valid
SELECT is_jsonb_valid_draft_v7('true', 'null');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- object is valid
SELECT is_jsonb_valid_draft_v7('true', '{"foo":"bar"}');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- empty object is valid
SELECT is_jsonb_valid_draft_v7('true', '{}');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- array is valid
SELECT is_jsonb_valid_draft_v7('true', '["foo"]');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- empty array is valid
SELECT is_jsonb_valid_draft_v7('true', '[]');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- boolean schema 'false'
-- number is invalid
SELECT is_jsonb_valid_draft_v7('false', '1');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- string is invalid
SELECT is_jsonb_valid_draft_v7('false', '"foo"');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- boolean true is invalid
SELECT is_jsonb_valid_draft_v7('false', 'true');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- boolean false is invalid
SELECT is_jsonb_valid_draft_v7('false', 'false');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- null is invalid
SELECT is_jsonb_valid_draft_v7('false', 'null');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- object is invalid
SELECT is_jsonb_valid_draft_v7('false', '{"foo":"bar"}');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- empty object is invalid
SELECT is_jsonb_valid_draft_v7('false', '{}');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- array is invalid
SELECT is_jsonb_valid_draft_v7('false', '["foo"]');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- empty array is invalid
SELECT is_jsonb_valid_draft_v7('false', '[]');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

